src/d3a/d3a_core/sim_results/market_price_energy_day.py

```diff
@@ -0,0 +1,111 @@
+
+        if area_dict['uuid'] not in fee_lists:
+            fee_lists[area_dict['uuid']] = OrderedDict()
+        if current_market_slot not in fee_lists[area_dict['uuid']].keys():
+            fee_lists[area_dict['uuid']][current_market_slot] = []
+        fee_rates = [
+            # Convert from cents to euro
+            t['fee_price'] / 100.0 / t['energy']
+            for t in core_stats.get(area_dict['uuid'], {}).get('trades', [])
+        ]
+        fee_lists[area_dict['uuid']][current_market_slot].extend(fee_rates)
+
+    def update(self, area_result_dict={}, core_stats={}, current_market_slot=None):
+        if current_market_slot is None:
+            return
+        current_price_lists, current_fee_lists = self.gather_trade_rates(
+            area_result_dict, core_stats, current_market_slot, {}, {}
+        )
+        price_energy_redis_output = {}
+        self._convert_output_format(current_price_lists, price_energy_redis_output,
+                                    current_fee_lists)
+        if self.should_export_plots:
+            self.calculate_csv_output(area_result_dict, price_energy_redis_output)
+        else:
+            self.redis_output = price_energy_redis_output
+
+    def calculate_csv_output(self, area_dict, price_energy_redis_output):
+        if not price_energy_redis_output.get(area_dict['uuid'], {}).get('price-energy-day', []):
+            return
+        if area_dict['name'] not in self.csv_output:
+            self.csv_output[area_dict['name']] = {
+                "price-currency": "Euros",
+                "load-unit": "kWh",
+                "price-energy-day": [],
+                "grid-fee-rate": []
+            }
+        self.csv_output[area_dict['name']]["price-energy-day"].\
+            append(price_energy_redis_output[area_dict['uuid']]['price-energy-day'])
+        self.csv_output[area_dict['name']]["grid-fee-rate"].\
+            append(price_energy_redis_output[area_dict['uuid']]['grid-fee-rate'])
+        for child in area_dict['children']:
+            self.calculate_csv_output(child, price_energy_redis_output)
+
+    @staticmethod
+    def _convert_output_format(price_energy, redis_output, fees):
+        for node_name, trade_rates in price_energy.items():
+            if node_name not in redis_output:
+                redis_output[node_name] = {
+                    "price-currency": "Euros",
+                    "load-unit": "kWh",
+                    "price-energy-day": [],
+                    "grid-fee-rate": []
+                }
+            redis_output[node_name]["price-energy-day"] = [
+                {
+                    "time": timeslot,
+                    "av_price": round_floats_for_ui(mean(trades) if len(trades) > 0 else 0),
+                    "min_price": round_floats_for_ui(min(trades) if len(trades) > 0 else 0),
+                    "max_price": round_floats_for_ui(max(trades) if len(trades) > 0 else 0),
```

Comment:
Would be far easier and more elegaant for the UI to add the grid fees as an additional value here.